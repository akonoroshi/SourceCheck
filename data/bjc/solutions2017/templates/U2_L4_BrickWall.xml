<project name="U2_L4_BrickWall" app="iSnap 2.1, http://go.ncsu.edu/isnap" version="1" guid="f64118b1-d737-482c-979b-db3c9498de60" assignment="U2_L4_BrickWall"><data>{}</data><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAAEQElEQVR4Xu2bvW4TQRRGJ0hIUBi5gIoCiZqfdHkCGn4qaAkPAEKEUMATQAMSJi9AkBAFPAEdVCmt8AikCC11MDISEtg4u/Pl27me5bj13O/aZ87OZmadlclkMkm8IBBEYAUBg8jT9hcBBESEUAIIGIqf5giIA6EEEDAUP80REAdCCSBgKH6aIyAOhBJAwFD8NEdAHAglgICh+GmOgDgQSgABQ/HTHAFxIJQAAobipzkC4kAoAQQMxU9zBMSBUAIIGIqf5giIA6EEEDAUP80REAdCCSBgKH6aIyAOhBJAwFD8NEdAHAglgICh+POaXz13Kb0df0rD4TCvcIlHVy3g6OyVVmjv731MOWOnoV2OV7P//B4Hx4+l8eVTabT1Kn249Sh9/7qfpu/X9kLAmRn7PYmqJE0CHPViuLj9JO2uP51r8+3kj3T67vX0YGOj6SMs1fsIWJmAF14/Tl/uPJuTaP/EQTpz7wYClry8clapnLG13IJ3V4fp5dYovb+5yS24pHj/a69r51fTu/HnNBgMeoOgV7fg6d9XOy/epJ3n240TtLa5ntYe3u50s1H6s7AJaZx274DZ2yoCsgv2GtaQhoB/H0OxAhbVb/6sjhWQFbCwgrSrnUCvNiGLJuOoh7+HTbJycN31kVBNUiJgZQfRbS4GBCxEIGclyRm7rAfRTVjZhDQRMr+fI1XOWAQ0T9QhcVXfgstholNXBKoWkHNAzgG7ujBa5SIgArYSpatBCIiAXbnVKhcBEbCVKAyCwCICvdqE8CSEZ8FFL/Wcs72csZwDlptGVkAexZWz7R+dEBABETCUAM1DCfRqBeQHqWxCil5NnANyDlhUuNlmCIiACLiAQMSfA/weMFRHmtdIoFebEJ6EsAkpehHmPN3IGcuTkHLTyArIQXQ523gS0sxa+TfL3BWz69W4+Vsuz4iqV8DlwcgnUQkgoEqOOgsBBLRgJEQlgIAqOeosBBDQgpEQlQACquSosxBAQAtGQlQCCKiSo85CAAEtGAlRCSCgSo46CwEEtGAkRCWAgCo56iwEENCCkRCVAAKq5KizEEBAC0ZCVAIIqJKjzkIAAS0YCVEJIKBKjjoLAQS0YCREJYCAKjnqLAQQ0IKREJUAAqrkqLMQQEALRkJUAgiokqPOQgABLRgJUQkgoEqOOgsBBLRgJEQlgIAqOeosBBDQgpEQlQACquSosxBAQAtGQlQCCKiSo85CAAEtGAlRCSCgSo46CwEEtGAkRCWAgCo56iwEENCCkRCVAAKq5KizEEBAC0ZCVAIIqJKjzkIAAS0YCVEJIKBKjjoLAQS0YCREJYCAKjnqLAQQ0IKREJUAAqrkqLMQQEALRkJUAgiokqPOQgABLRgJUQkgoEqOOgsBBLRgJEQlgIAqOeosBBDQgpEQlQACquSosxBAQAtGQlQCCKiSo85CAAEtGAlRCSCgSo46C4GfRj5o1aeNjjIAAAAASUVORK5CYII=</thumbnail><stage name="Stage" guid="f64118b1-d737-482c-979b-db3c9498de60" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="flat" codify="false" inheritance="false" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAUcklEQVR4Xu3dwZEktxGG0R43RHdEe9YELk1Yeyh3KDdG141QLyP6z0QngHk8b1ahHtD11WEi+PHwHwECBAgQIPB2gY+339ENCRAgQIAAgYcAOwQECBAgQGBAQIAH0N2SAAECBAgIsDNAgAABAgQGBAR4AN0tCRAgQICAADsDBAgQIEBgQECAB9DdkgABAgQICLAzQIAAAQIEBgQEeADdLQkQIECAgAA7AwQIECBAYEBAgAfQ3ZIAAQIECAiwM0CAAAECBAYEBHgA3S0JECBAgIAAOwMECBAgQGBAQIAH0N2SAAECBAgIsDNAgAABAgQGBAR4AN0tCRAgQICAADsDBAgQIEBgQECAB9DdkgABAgQICLAzQIAAAQIEBgQEeADdLQkQIECAgAA7AwQIECBAYEBAgAfQ3ZIAAQIECAiwM0CAAAECBAYEBHgA3S0JECBAgIAAOwMECBAgQGBAQIAH0N2SAAECBAgIsDNAgAABAgQGBAR4AN0tCRAgQICAADsDBAgQIEBgQECAB9DdkgABAgQICLAzQIAAAQIEBgQEeADdLQkQIECAgAA7AwQIECBAYEBAgAfQ3ZIAAQIECAiwM0CAAAECBAYEBHgA3S0JECBAgIAAOwMECBAgQGBAQIAH0N2SAAECBAgIsDNAgAABAgQGBAR4AN0tCRAgQICAADsDBAgQIEBgQECAB9DdkgABAgQICLAzQIAAAQIEBgQEeADdLQkQIECAgAA7AwQIECBAYEBAgAfQ3ZIAAQIECAiwM0CAAAECBAYEBHgA3S0JECBAgIAAOwMECBAgQGBAQIAH0N2SAAECBAgIsDNAgAABAgQGBAR4AN0tCRAgQICAADsDBAgQIEBgQECAB9DdkgABAgQICLAzQIAAAQIEBgQEeADdLQkQIECAgAA7AwQIECBAYEBAgAfQ3ZIAAQIECAiwM0CAAAECBAYEBHgA3S0JECBAgIAAOwMECBAgQGBAQIAH0N2SAAECBAgIsDNAgAABAgQGBAR4AN0tCRAgQICAADsDBAgQIEBgQECAB9DdkgABAgQICLAzQIAAAQIEBgQEeADdLQkQIECAgAA7AwQIECBAYEBAgAfQ3ZIAAQIECAiwM0CAAAECBAYEBHgA3S0JECBAgIAAOwMECBAgQGBAQIAH0N2SAAECBAgIsDNAgAABAgQGBAR4AN0tCRAgQICAADsDBAgQIEBgQECAB9DdkgABAgQICLAzQIAAAQIEBgQEeADdLQkQIECAgAA7AwQIECBAYEBAgAfQ3ZIAAQIECAiwM0CAAAECBAYEBHgA/eRb/vjXvz8n1v/tv//5v7NqLY8Hl+encXeXid+Qe+4nIMD77cnWKxI90fvVAd09ejud3a1/5Bb3NgEBfhv1HTfa6SVmLT4GTv0YuONt4CmqAgJcFfxi86IneqdGb6ez+8VeGx73FwIC7Gi8JLDTS8xafAyc+jHw0o/OP75WQICv3do1DyZ6ondq9HY6u2t+na56moAAn7Zjw+vd6SVmLT4GTv0YGP4Zu/0mAgK8yUacsgzRE71To7fT2T3l926dawUEeK3vdVff6SVmLT4GTv0YuO7F4IEiAQGO2L7ukOiJ3qnR2+nsft03iCf/WUCAnQcCBAgQIDAgIMAD6G5JgAABAgQE2BkgQIAAAQIDAgI8gO6WBAgQIEBAgJ0BAgQIECAwICDAA+gn3/LlvyT9+Pj+7e+//vz5mX/89vsfj8/P7293sJbn5FyWujz7v0S9/ey74ZYCArzltuy7KAFu2hvRWxq9ll1q2iMBbtmNKy8iwFdu67qHEuAm26aXe8tqrGXpx4AAt5zSKy8iwFdu67qHEuAmW9FbGr2WXWraIwFu2Y0rLyLAV27ruocS4Cbbppd7y2qsZenHgAC3nNIrLyLAV27ruocS4CZb0VsavZZdatojAW7ZjSsvIsBXbuu6hxLgJtuml3vLaqxl6ceAALec0isvIsBXbuu6hxLgJlvRWxq9ll1q2iMBbtmNKy8iwFdu67qHEuAm26aXe8tqrGXpx4AAt5zSKy8iwFdu67qHEuAmW9FbGr2WXWraIwFu2Y0rLyLAV26rhyJAgACB3QUEePcdsj4CBAgQuFJAgK/cVg9FgAABArsLCPDuO2R9BAgQIHClgABfua0eigABAgR2FxDg3Xdos/W9/FfQTet/9pek1vJ4cHl+wHZ3afpZuMzhAgJ8+Aa+e/miJ3q/OnO7R2+ns/vu36377SkgwHvuy7ar2uklZi0+Bk79GNj2B25hbxUQ4Ldyn38z0RO9U6O309k9/03gCToEBLhD8QtdY6eXmLX4GDj1Y+ALvTI86j8ICLDj8ZKA6IneqdHb6ey+9KPzj68VEOBrt3bNg+30ErMWHwOnfgys+XW66mkCAnzajg2vV/RE79To7XR2h3/Gbr+JgABvshGnLGOnl5i1+Bg49WPglN+7da4VEOC1vtddXfRE79To7XR2r3sxeKBIQIAjNkMECBAgQKAmIMA1P9MECBAgQCASEOCIzRABAgQIEKgJCHDNzzQBAgQIEIgEBDhiM0SAAAECBGoCAlzz+3LTL/8l6cfH929///Xnz1A/fvv9j8fn5/e341nLc3IuS12e/V+i3n723XBLAQHeclv2XZQAN+2N6C2NXssuNe2RALfsxpUXEeArt3XdQwlwk23Ty71lNday9GNAgFtO6ZUXEeArt3XdQwlwk63oLY1eyy417ZEAt+zGlRcR4Cu3dd1DCXCTbdPLvWU11rL0Y0CAW07plRcR4Cu3dd1DCXCTregtjV7LLjXtkQC37MaVFxHgK7d13UMJcJNt08u9ZTXWsvRjQIBbTumVFxHgK7d13UMJcJOt6C2NXssuNe2RALfsxpUXEeArt3XdQwlwk23Ty71lNday9GNAgFtO6ZUXEeArt3XdQwlwk63oLY1eyy417ZEAt+zGlRcR4Cu31UMRIECAwO4CArz7DlkfAQIECFwpIMBXbquHIkCAAIHdBQR49x2yPgIECBC4UkCAr9xWD0WAAAECuwsI8O47tNn6Xv4r6Kb1P/tLUmt5PLg8P2C7uzT9LFzmcAEBPnwD37180RO9X5253aO309l99+/W/fYUEOA992XbVe30ErMWHwOnfgxs+wO3sLcKCPBbuc+/meiJ3qnR2+nsnv8m8AQdAgLcofiFrrHTS8xafAyc+jHwhV4ZHvUfBATY8XhJQPRE79To7XR2X/rR+cfXCgjwtVu75sF2eolZi4+BUz8G1vw6XfU0AQE+bceG1yt6ondq9HY6u8M/Y7ffRECAN9mIU5ax00vMWnwMnPoxcMrv3TrXCgjwWt/rri56ondq9HY6u9e9GDxQJCDAEZshAgQIECBQExDgmp9pAgQIECAQCQhwxGaIAAECBAjUBAS45meaAAECBAhEAgIcsRkiQIAAAQI1AQGu+X256Zf/kvTj4/u3v//682eoH7/9/sfj8/P72/Gs5Tk5l6Uuz/4vUW8/+264pYAAb7kt+y5KgJv2RvSWRq9ll5r2SIBbduPKiwjwldu67qEEuMm26eXeshprWfoxIMAtp/TKiwjwldu67qEEuMlW9JZGr2WXmvZIgFt248qLCPCV27ruoQS4ybbp5d6yGmtZ+jEgwC2n9MqLCPCV27ruoQS4yVb0lkavZZea9kiAW3bjyosI8JXbuu6hBLjJtunl3rIaa1n6MSDALaf0yosI8JXbuu6hBLjJVvSWRq9ll5r2SIBbduPKiwjwldu67qEEuMm26eXeshprWfoxIMAtp/TKiwjwldu67qEEuMlW9JZGr2WXmvZIgFt248qLCPCV2+qhCBAgQGB3AQHefYesjwABAgSuFBDgK7fVQxEgQIDA7gICvPsOWR8BAgQIXCkgwFduq4ciQIAAgd0FBHj3HdpsfS//FXTT+p/9Jam1PB5cnh+w3V2afhYuc7iAAB++ge9evuiJ3q/O3O7R2+nsvvt36357Cgjwnvuy7ap2eolZi4+BUz8Gtv2BW9hbBQT4rdzn30z0RO/U6O10ds9/E3iCDgEB7lD8QtfY6SVmLT4GTv0Y+EKvDI/6DwIC7Hi8JCB6ondq9HY6uy/96PzjawUE+NqtXfNgO73ErMXHwKkfA2t+na56moAAn7Zjw+sVPdE7NXo7nd3hn7HbbyIgwJtsxCnL2OklZi0+Bk79GDjl926dawUEeK3vdVcXPdE7NXo7nd3rXgweKBIQ4IjNEAECBAgQqAkIcM3PNAECBAgQiAQEOGIzRIAAAQIEagICXPMzTYAAAQIEIgEBjtgMESBAgACBmoAA1/xMEyBAgACBSECAIzZDBAgQIECgJiDANT/TBAgQIEAgEhDgiM0QAQIECBCoCQhwzc80AQIECBCIBAQ4YjNEgAABAgRqAgJc8zNNgAABAgQiAQGO2AwRIECAAIGagADX/EwTIECAAIFIQIAjNkMECBAgQKAmIMA1P9MECBAgQCASEOCIzRABAgQIEKgJCHDNzzQBAgQIEIgEBDhiM0SAAAECBGoCAlzzM02AAAECBCIBAY7YDBEgQIAAgZqAANf8TBMgQIAAgUhAgCM2QwQIECBAoCYgwDU/0wQIECBAIBIQ4IjNEAECBAgQqAkIcM3PNAECBAgQiAQEOGIzRIAAAQIEagICXPMzTYAAAQIEIgEBjtgMESBAgACBmoAA1/xMEyBAgACBSECAIzZDBAgQIECgJiDANT/TBAgQIEAgEhDgiM0QAQIECBCoCQhwzc80AQIECBCIBAQ4YjNEgAABAgRqAgJc8zNNgAABAgQiAQGO2AwRIECAAIGagADX/EwTIECAAIFIQIAjNkMECBAgQKAmIMA1P9MECBAgQCASEOCIzRABAgQIEKgJCHDNzzQBAgQIEIgEBDhiM0SAAAECBGoCAlzzM02AAAECBCIBAY7YDBEgQIAAgZqAANf8TBMgQIAAgUhAgCM2QwQIECBAoCYgwDU/0wQIECBAIBIQ4IjNEAECBAgQqAkIcM3PNAECBAgQiAQEOGIzRIAAAQIEagICXPMzTYAAAQIEIgEBjtgMESBAgACBmoAA1/xMEyBAgACBSECAIzZDBAgQIECgJiDANT/TBAgQIEAgEhDgiM0QAQIECBCoCQhwzc80AQIECBCIBAQ4YjNEgAABAgRqAgJc8zNNgAABAgQiAQGO2AwRIECAAIGagADX/EwTIECAAIFIQIAjNkMECBAgQKAmIMA1P9MECBAgQCASEOCIzRABAgQIEKgJCHDNzzQBAgQIEIgEBDhiM0SAAAECBGoCAlzzM02AAAECBCIBAY7YDBEgQIAAgZqAANf8TBMgQIAAgUhAgCM2QwQIECBAoCYgwDU/0wQIECBAIBIQ4IjNEAECBAgQqAkIcM3PNAECBAgQiAQEOGIzRIAAAQIEagICXPMzTYAAAQIEIgEBjtgMESBAgACBmoAA1/xMEyBAgACBSECAIzZDBAgQIECgJiDANT/TBAgQIEAgEhDgiM0QAQIECBCoCQhwzc80AQIECBCIBAQ4YjNEgAABAgRqAgJc8zNNgAABAgQiAQGO2AwRIECAAIGagADX/EwTIECAAIFIQIAjNkMECBAgQKAmIMA1P9MECBAgQCASEOCIzRABAgQIEKgJCHDNzzQBAgQIEIgEBDhiM0SAAAECBGoCAlzzM02AAAECBCIBAY7YDBEgQIAAgZqAANf8TBMgQIAAgUhAgCM2QwQIECBAoCYgwDU/0wQIECBAIBIQ4IjNEAECBAgQqAkIcM3PNAECBAgQiAQEOGIzRIAAAQIEagICXPMzTYAAAQIEIgEBjtgMESBAgACBmoAA1/xMEyBAgACBSECAIzZDBAgQIECgJiDANT/TBAgQIEAgEhDgiM0QAQIECBCoCQhwzc80AQIECBCIBAQ4YjNEgAABAgRqAgJc8zNNgAABAgQiAQGO2AwRIECAAIGagADX/EwTIECAAIFIQIAjNkMECBAgQKAmIMA1P9MECBAgQCASEOCIzRABAgQIEKgJCHDNzzQBAgQIEIgEBDhiM0SAAAECBGoCAlzzM02AAAECBCIBAY7YDBEgQIAAgZqAANf8TBMgQIAAgUhAgCM2QwQIECBAoCYgwDU/0wQIECBAIBIQ4IjNEAECBAgQqAkIcM3PNAECBAgQiAQEOGIzRIAAAQIEagICXPMzTYAAAQIEIgEBjtgMESBAgACBmoAA1/xMEyBAgACBSECAIzZDBAgQIECgJiDANT/TBAgQIEAgEhDgiM0QAQIECBCoCQhwzc80AQIECBCIBAQ4YjNEgAABAgRqAgJc8zNNgAABAgQiAQGO2AwRIECAAIGagADX/EwTIECAAIFIQIAjNkMECBAgQKAmIMA1P9MECBAgQCASEOCIzRABAgQIEKgJCHDNzzQBAgQIEIgEBDhiM0SAAAECBGoCAlzzM02AAAECBCIBAY7YDBEgQIAAgZqAANf8TBMgQIAAgUhAgCM2QwQIECBAoCYgwDU/0wQIECBAIBIQ4IjNEAECBAgQqAkIcM3PNAECBAgQiAQEOGIzRIAAAQIEagICXPMzTYAAAQIEIgEBjtgMESBAgACBmoAA1/xMEyBAgACBSECAIzZDBAgQIECgJiDANT/TBAgQIEAgEhDgiM0QAQIECBCoCQhwzc80AQIECBCIBAQ4YjNEgAABAgRqAgJc8zNNgAABAgQiAQGO2AwRIECAAIGagADX/EwTIECAAIFIQIAjNkMECBAgQKAmIMA1P9MECBAgQCASEOCIzRABAgQIEKgJCHDNzzQBAgQIEIgEBDhiM0SAAAECBGoCAlzzM02AAAECBCIBAY7YDBEgQIAAgZqAANf8TBMgQIAAgUhAgCM2QwQIECBAoCYgwDU/0wQIECBAIBIQ4IjNEAECBAgQqAkIcM3PNAECBAgQiAQEOGIzRIAAAQIEagICXPMzTYAAAQIEIgEBjtgMESBAgACBmoAA1/xMEyBAgACBSECAIzZDBAgQIECgJvA/9Bol8OQo+tcAAAAASUVORK5CYII=</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><sprite name="Sprite" idx="1" x="110" y="90" heading="90" scale="1" rotation="1" draggable="true" costume="0" color="145,26,68" pen="tip" id="8"><costumes><list id="9"></list></costumes><sounds><list id="10"></list></sounds><variables></variables><blocks></blocks><scripts><script x="210" y="146"><custom-block id="244" s="row a"></custom-block></script><script x="250" y="208"><custom-block id="245" s="row b"></custom-block></script><script x="393" y="135"><block id="246" s="clear"></block><custom-block id="247" s="draw brick wall with %s rows"><l>7</l></custom-block></script></scripts></sprite></sprites></stage><hidden></hidden><headers></headers><code></code><blocks><block-definition s="set flag %&apos;which&apos; to %&apos;tf&apos;" type="command" category="sensing" guid="7748826d-e1ab-461b-ab45-5a18b9d9b6f4" isImported="true"><header></header><code></code><inputs><input type="%s" readonly="true"><options>Presentation mode
Retina display support
Long form input dialog
Plain prototype labels
Input sliders
Execute on slider change
Clicking sound
Turbo mode
Flat design
Keyboard editing
Table support
Table lines
Visible stepping
Thread safe scripts
Prefer smooth animations
Flat line ends
Codification support
Inheritance support</options></input><input type="%b"></input></inputs><script><block id="198" s="doRun"><block id="200" s="reportJSFunction"><list><l>which</l><l>tf</l></list><l>var   stage = this.parentThatIsA(StageMorph),&#xD;        ide = stage.parentThatIsA(IDE_Morph),&#xD;        world = stage.parentThatIsA(WorldMorph);&#xD;&#xD;if (tf != !!tf) return;&#xD;&#xD;try{&#xD;ide.savingPreferences = false;&#xD;&#xD;switch (which) {&#xD;    case &apos;Presentation mode&apos;:&#xD;        if (tf != ide.isAppMode) ide.toggleAppMode();&#xD;        break;&#xD;    case &apos;Retina display support&apos;:&#xD;        if (tf != isRetinaEnabled()) ide.toggleRetina();&#xD;        break;&#xD;    case &apos;Long form input dialog&apos;:&#xD;        if (tf != InputSlotDialogMorph.prototype.isLaunchingExpanded)&#xD;            ide.toggleLongFormInputDialog();&#xD;        break;&#xD;    case &apos;Plain prototype labels&apos;:&#xD;        if (tf != BlockLabelPlaceHolderMorph.prototype.plainLabel)&#xD;            ide.togglePlainPrototypeLabels();&#xD;        break;&#xD;    case &apos;Input sliders&apos;:&#xD;        MorphicPreferences.useSliderForInput = tf;&#xD;        break;&#xD;    case &apos;Execute on slider change&apos;:&#xD;        ArgMorph.prototype.executeOnSliderEdit = tf;&#xD;        break;&#xD;    case &apos;Clicking sound&apos;:&#xD;        if (tf != !!BlockMorph.prototype.snapSound)&#xD;            BlockMorph.prototype.toggleSnapSound();&#xD;        break;&#xD;    case &apos;Turbo mode&apos;:&#xD;        if (tf != stage.isFastTracked)&#xD;            ide.toggleFastTracking();&#xD;        break;&#xD;    case &apos;Flat design&apos;:&#xD;        if (tf == MorphicPreferences.isFlat) break;&#xD;        if (tf)&#xD;            ide.flatDesign();&#xD;        else&#xD;            ide.defaultDesign();&#xD;        break;&#xD;    case &apos;Keyboard editing&apos;:&#xD;            ScriptsMorph.prototype.enableKeyboard = tf;&#xD;        break;&#xD;    case &apos;Table support&apos;:&#xD;            List.prototype.enableTables = tf;&#xD;        break;&#xD;    case &apos;Table lines&apos;:&#xD;                TableMorph.prototype.highContrast = tf;&#xD;        break;&#xD;    case &apos;Visible stepping&apos;:&#xD;        if (tf != Process.prototype.enableSingleStepping)&#xD;            ide.toggleSingleStepping();&#xD;        break;&#xD;    case &apos;Thread safe scripts&apos;:&#xD;        stage.isThreadSafe = tf;&#xD;        break;&#xD;    case &apos;Prefer smooth animations&apos;:&#xD;        if (tf != (StageMorph.prototype.frameRate &gt; 0))&#xD;            ide.toggleVariableFrameRate();&#xD;        break;&#xD;    case &apos;Flat line ends&apos;:&#xD;        SpriteMorph.prototype.useFlatLineEnds = tf;&#xD;        break;&#xD;    case &apos;Codification support&apos;:&#xD;        if (tf != StageMorph.prototype.enableCodeMapping) {&#xD;            StageMorph.prototype.enableCodeMapping = tf;&#xD;            ide.currentSprite.blocksCache.variables = null;&#xD;            ide.currentSprite.paletteCache.variables = null;&#xD;            ide.refreshPalette();&#xD;        }&#xD;        break;&#xD;    case &apos;Inheritance support&apos;:&#xD;        if (tf != StageMorph.prototype.enableInheritance) {&#xD;            StageMorph.prototype.enableInheritance = tf;&#xD;            ide.currentSprite.blocksCache.variables = null;&#xD;            ide.currentSprite.paletteCache.variables = null;&#xD;            ide.refreshPalette();&#xD;        }&#xD;        break;&#xD;};&#xD;}&#xD;finally {&#xD;ide.savingPreferences = false;&#xD;};&#xD;</l></block><list><block id="201" var="which"/><block id="202" var="tf"/></list></block></script></block-definition><block-definition s="draw brick, length: %&apos;length&apos; width: %&apos;width&apos;" type="command" category="pen" guid="b22a01ed-eff2-4bee-b2ff-1ab9299d8a27" isImported="false"><header></header><code></code><inputs><input type="%n"></input><input type="%n"></input></inputs><script><block id="203" s="setColor"><color>145,26,68,1</color></block><custom-block id="204" s="set flat line ends to %b"><block id="205" s="reportBoolean"><l><bool>true</bool></l></block></custom-block><block id="206" s="setSize"><block id="207" var="width"/></block><block id="208" s="down"></block><block id="209" s="forward"><block id="210" var="length"/></block><block id="211" s="up"></block></script></block-definition><block-definition s="set flat line ends to %&apos;flag&apos;" type="command" category="pen" guid="3f4ec97a-8371-4099-83b9-886fd2322b2c" isImported="false"><header></header><code></code><inputs><input type="%b"></input></inputs><script><custom-block id="212" s="set flag %s to %b"><l>Flat line ends</l><l><bool>true</bool></l></custom-block></script></block-definition><block-definition s="row a" type="command" category="pen" guid="56a41c4e-c100-40f6-852e-f9f72eb2e316" isImported="false"><header></header><code></code><inputs></inputs><script><block id="213" s="doRepeat"><l>6</l><script><custom-block id="214" s="draw brick, length: %n width: %n"><l>30</l><l>10</l></custom-block><block id="215" s="forward"><l>5</l></block></script></block></script></block-definition><block-definition s="row b" type="command" category="pen" guid="5c193512-db22-4ad0-9845-7b55b7c9a826" isImported="false"><header></header><code></code><inputs></inputs><script><custom-block id="216" s="draw brick, length: %n width: %n"><l>12.5</l><l>10</l></custom-block><block id="217" s="forward"><l>5</l></block><block id="218" s="doRepeat"><l>5</l><script><custom-block id="219" s="draw brick, length: %n width: %n"><l>30</l><l>10</l></custom-block><block id="220" s="forward"><l>5</l></block></script></block><custom-block id="221" s="draw brick, length: %n width: %n"><l>12.5</l><l>10</l></custom-block></script></block-definition><block-definition s="draw brick wall with %&apos;row&apos; rows" type="command" category="pen" guid="7e197b7d-266b-43b7-9102-8823bda5ba7a" isImported="false"><header></header><code></code><inputs><input type="%s"></input></inputs><script><block id="222" s="doDeclareVariables"><list><l>i</l></list></block><block id="225" s="doRepeat"><block id="226" var="row"/><script><block id="227" s="gotoXY"><l>-100</l><block id="228" s="reportProduct"><block id="229" var="i"/><l>15</l></block></block><block id="230" s="doIfElse"><custom-block id="231" s="is even %s"><block id="232" var="i"/></custom-block><script><custom-block id="233" s="row a"></custom-block></script><script><custom-block id="234" s="row b"></custom-block></script></block><block id="235" s="doChangeVar"><l>i</l><l>1</l></block></script></block></script></block-definition><block-definition s="is even %&apos;n&apos;" type="predicate" category="operators" guid="4d0e3151-4d66-4c76-83d6-2a2fa0b54255" isImported="false"><header></header><code></code><inputs><input type="%s"></input></inputs><script><block id="236" s="doReport"><custom-block id="237" s="%s divisible by %s"><block id="238" var="n"/><l>2</l></custom-block></block></script></block-definition><block-definition s="%&apos;x&apos; divisible by %&apos;y&apos;" type="predicate" category="operators" guid="f8bf89e8-51ef-4632-95b3-a4e488639be6" isImported="false"><header></header><code></code><inputs><input type="%s"></input><input type="%s"></input></inputs><script><block id="239" s="doReport"><block id="240" s="reportEquals"><block id="241" s="reportModulus"><block id="242" var="x"/><block id="243" var="y"/></block><l>0</l></block></block></script></block-definition></blocks><variables></variables><editing></editing></project>