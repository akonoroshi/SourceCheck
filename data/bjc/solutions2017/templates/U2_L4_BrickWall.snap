snapshot {
    @defVar GLOBAL {
        stage {
            sprite {
                @repeat (1, 4) {
                    script {
                        evaluateCustomBlock(literal)
                    }
                }
            }
        }
        customBlock {
            script {
                doRun(reportJSFunction(list(literal, literal), literal), list(var, var))
            }
            var
            var
        }
        customBlock {
            script {
                setColor(literal)
                evaluateCustomBlock(reportBoolean(literal))
                setSize(var)
                down
                forward(var)
                up
            }
            var
            var
        }
        customBlock {
            script {
                evaluateCustomBlock(literal, literal)
            }
            var
        }
        customBlock {
            script {
                doRepeat(literal, script {
                    evaluateCustomBlock(literal, literal)
                    forward(literal)
                })
            }
        }
        customBlock {
            script {
                evaluateCustomBlock(literal, literal)
                @or {
                    @inline {
                        forward(literal)
                        doRepeat(literal, script {
                            evaluateCustomBlock(literal, literal)
                            forward(literal)
                        })
                    }
                    @inline {
                        doRepeat(literal, script {
                            forward(literal)
                            evaluateCustomBlock(literal, literal)
                        })
                        forward(literal)
                    }
                }
                evaluateCustomBlock(literal, literal)
            }
        }
        @defVar (IMPORT) {
            @if (IMPORT) {
                customBlock {
                    script {
                        doReport(evaluateCustomBlock(var, literal))
                    }
                    var
                }
                customBlock {
                    script {
                        doReport(reportEquals(literal, reportModulus(var, var)))
                    }
                    var
                    var
                }
            }
            customBlock {
                script {
                    @defVar (MULT) {
                        @anyOrder {
                            @if (GLOBAL) { doSetVariable(literal, literal) }
                            @unless (GLOBAL) { doDeclareVariables(list(literal)) }
                            @unless (MULT) {
                                @or {
                                    setYPosition(literal)
                                    gotoXY(literal, literal)
                                }
                            }
                        }
                        doRepeat(var, script {
                            @unless (MULT) {
                                setXPosition(literal)
                                changeYPosition(literal)
                            }
                            @if (MULT) {
                                gotoXY(literal, @or {
                                    reportProduct(literal, var)
                                    reportSum(literal, reportProduct(literal, var))
                                })
                            }
                            doIfElse(
                                @if (IMPORT) { evaluateCustomBlock(var) }
                                @unless (IMPORT)  {
                                    reportEquals(literal, reportModulus(var, literal))
                                }
                                script {
                                    evaluateCustomBlock
                                }, script {
                                    evaluateCustomBlock
                                }
                            )
                            doChangeVar(literal, literal)
                        })
                    }
                }
                var
            }
        }
        @if (GLOBAL) { var }
    }
}